app/scripts/lib/extension-port-stream.ts

```diff
@@ -0,0 +1,252 @@
+import type { Json } from '@metamask/utils';
```

Comment:
Todo: FF uses structuredClone for port messaging, which allows buffers. MV2 supports passing Blob URLs, which is probably better than anything else

Reply:
Update: Blob URLs were slow.

Reply:
I also tested passing an ArrayBuffer (on FF, ArrayBuffers don't work on Chrome) and it was like 2% faster for very large messages, but the overhead of creating it from a JSON object would cancel this out. So it's not worth it.

Also, FF doesn't have a port limit... so it doesn't matter anyway.